app/src/main/java/com/nervousfish/nervousfish/service_locator/NervousFish.java

```diff
@@ -10,86 +10,84 @@
 import com.nervousfish.nervousfish.modules.pairing.AndroidBluetoothService;
 import com.nervousfish.nervousfish.modules.pairing.PairingWrapper;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /**
  * The controller of the NervousFish application. It provides an access point for the
- * {@link com.nervousfish.nervousfish.modules.pairing.IBluetoothHandlerService}
- * and holds the current global state of the application.
+ * {@link com.nervousfish.nervousfish.modules.pairing.IBluetoothHandlerService} and holds the
+ * current global state of the application.
  */
 public final class NervousFish extends Application implements INervousFish {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("NervousFish");
+
     private static NervousFish instance;
+    private static IServiceLocator serviceLocator;
 
+    private final ServiceConnection connection = new BluetoothServiceConnection();
     private AndroidBluetoothService bluetoothService;
-    private boolean bound;
-    private Runnable onServiceBoundRunnable;
-    private final ServiceConnection connection = new ServiceConnectionImpl();
-
-    /**
-     * @return The NervousFish {@link Application} instance
-     */
-    public static Context getInstance() {
-        return instance;
-    }
 
     /**
-     * {@inheritDoc     }
+     * {@inheritDoc}
      */
     @Override
     public void onCreate() {
         super.onCreate();
 
         final Intent serviceIntent = new Intent(this, AndroidBluetoothService.class);
-        bindService(serviceIntent, connection, Context.BIND_AUTO_CREATE);
+        this.bindService(serviceIntent, this.connection, Context.BIND_AUTO_CREATE);
 
+        final String androidFileDir = this.getFilesDir().getPath();
+        NervousFish.serviceLocator = new ServiceLocator(androidFileDir);
```

Comment:
> Is there a way to use another service locator in the tests if the need arises somewhere in the future?

Not that I know/can think off..


> However, just thinking, maybe it is very good practice to do that checking explicitly

Agreed 😄 Also, we have some very basic checks for Bluetooth but I'm not sure how good that works...